ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/RecoveryTest.java

```diff
@@ -707,9 +704,42 @@ public void shouldRecoverQueryIDs() {
     final Set<QueryId> queryIdNames = queriesById(server.ksqlEngine.getPersistentQueries())
         .keySet();
 
-    assertThat(queryIdNames, contains(new QueryId("CSAS_C_0")));
+    assertThat(queryIdNames, contains(new QueryId("CSAS_C_1")));
   }
 
+  @Test
+  public void shouldIncrementQueryIDsNoPlans() {
+    server1.submitCommands(
+        "CREATE STREAM A (COLUMN STRING) WITH (KAFKA_TOPIC='A', VALUE_FORMAT='JSON');",
+        "CREATE STREAM B AS SELECT * FROM A;",
+        "TERMINATE CSAS_B_1;");
+
+    final KsqlServer server = new KsqlServer(commands);
+    server.recover();
+    server.submitCommands("CREATE STREAM C AS SELECT * FROM A;");
+    final Set<QueryId> queryIdNames = queriesById(server.ksqlEngine.getPersistentQueries())
+        .keySet();
+
+    assertThat(queryIdNames, contains(new QueryId("CSAS_C_2")));
+  }
+
+  @Test
+  public void shouldIncrementQueryIDsWithPlan() {
```

Comment:
I feel like I'm blind, what's the difference between these two tests?

Reply:
No you are not blind, I messes it up somehow. Fixed it though